support-workers/src/main/scala/com/gu/support/workers/lambdas/SendThankYouEmail.scala

```diff
@@ -61,11 +61,11 @@ class SendThankYouEmail(servicesProvider: ServiceProvider = ServiceProvider)
             amount = c.amount,
             paymentMethod = paymentMethodWithSchedule.paymentMethod
           )
-        )
-      case _: DigitalPack => Right(DigitalPackEmailFields.build(
-        getSubscriptionEmailFields(state, directDebitMandateId),
+        ).map(List(_))
+      case d: DigitalPack => new DigitalPackEmailFields(getSubscriptionEmailFields(state, directDebitMandateId)).build(
         paidSubPaymentData = state.paymentOrRedemptionData.left.toOption,
-      ))
+        readerType = d.readerType
+      )
```

Comment:
the builder may return a Left either if we try to send a redemption email for a non corp/gift sub, or if the ADT structure could not be serialised to a flat key/value pairs list.
Because the Either is sequenced this lambda should still be idempodent, so if we want to send multiple emails, and one fails, it will refuse to send either.  This means if it is retrying repeatedy, the gifter wouldn't receive multiple emails!